code/game/turfs/walls/wall_natural.dm

```diff
@@ -65,11 +66,16 @@
 
 // Drill out natural walls.
 /turf/wall/natural/handle_wall_tool_interactions(obj/item/W, mob/user)
-	if(IS_PICK(W))
+	if(IS_PICK(W) && !being_mined)
 		if(W.material?.hardness < max(material?.hardness, reinf_material?.hardness))
 			to_chat(user, SPAN_WARNING("\The [W] is not hard enough to dig through \the [src]."))
-		else if(W.do_tool_interaction(TOOL_PICK, user, src, 2 SECONDS, suffix_message = destroy_artifacts(W, INFINITY)))
-			dismantle_wall()
+		else
+			if(being_mined)
+				return TRUE
+			being_mined = TRUE
+			if(W.do_tool_interaction(TOOL_PICK, user, src, 2 SECONDS, suffix_message = destroy_artifacts(W, INFINITY)))
```

Comment:
I believe that some things use `set_cooldown = TRUE` instead of a bespoke debounce var, but that *does* mean you still have the cooldown even when you cancel it, and additional vars have a constant (not scaling with number of instances) cost anyway.

Reply:
Worse yet, it would mean bump-mining would still have no cooldown. Good call.